std/haxe/coro/continuations/RacingContinuation.hx

```diff
@@ -30,15 +30,15 @@ private class Thread {
 
 @:coreApi class RacingContinuation<T> implements IContinuation<T> {
 	final inputCont:IContinuation<Any>;
```

Comment:
I wonder if this should get its own type parameter. That would also mean that `suspend` gets a second one, which may or may not make sense.

Reply:
I _think_ it makes sense, at the very least the continuation passed into the `coroutine` lambda should be able to be `T` now instead of `Any`.

Reply:
Right, I've changed that.

And now I'm not sure anymore if there are even two different types involved. I thought there were but I can't really tell why. `RacingContinuation.resume` mixes this all together by using `result` as an argument to `inputCont.resume` and by assigning it to `outputCont._hx_result`, which suggests that they're actually the same type.

Reply:
I've just been trying to figure that one out as well, it is quite suspicious that changing the other `Any` involved there to `T` doesn't break anything (for the cpp and jvm tests atleast).
Still not figured out if this is correct or something we're not testing is subtly broken...

Reply:
I think I keep confusing myself because in the general case, the call to a suspend function can of course give a different type than what the caller will ultimately return. But this is about `suspend` in particular, which kind of inserts itself into the whole chain. So if we originally have a coro which goes `A -> B`, with a `suspend` we basically go `A -> (B -> B)` and so it seems sensible that the types would match.

... but all this is more guesswork than anything else. I've pushed a change that uses `T` for both the input and output continuations. If that turns out to be wrong after all we can pretty easily change it later because this doesn't affect the internal typing in any way.